presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewLogicalPlanner.java

```diff
@@ -1077,6 +1089,551 @@ public void testBaseToViewConversionWithGroupBy()
+        }
+    }
+
+    @Test
+    public void testBaseToViewConversionCountOutputAsExpected()
+    {
+        Session queryOptimizationWithMaterializedView = Session.builder(getSession())
+                .setSystemProperty(QUERY_OPTIMIZATION_WITH_MATERIALIZED_VIEW_ENABLED, "true")
+                .build();
+        QueryRunner queryRunner = getQueryRunner();
+
+        String table = "my_table";
+        String view = "my_view";
+
+        try {
+            queryRunner.execute(format("CREATE TABLE %s" +
+                            "(a_field bigint, grouping_key bigint, ds varchar(10)) " +
+                            "WITH (partitioned_by=array['ds'])",
+                    table));
+
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s " +
+                            "WITH (partitioned_by = ARRAY['ds']) " +
+                            "AS SELECT COUNT(*) AS a_count, grouping_key, ds " +
+                            "FROM %s " +
+                            "GROUP BY ds, grouping_key\n",
+                    view, table));
+
+            assertTrue(getQueryRunner().tableExists(getSession(), view));
+
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (1, 10, '2021-01-01')", table), 1);
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (2, 11, '2021-01-01')", table), 1);
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (3, 12, '2021-01-01')", table), 1);
+
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (4, 13, '2021-01-02')", table), 1);
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (5, 10, '2021-01-02')", table), 1);
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (6, 10, '2021-01-02')", table), 1);
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
+                    "VALUES (7, 10, '2021-01-02')", table), 1);
+
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE ds = '2021-01-01'", view), 3);
+
+            String viewQuery = format("SELECT SUM(a_count), grouping_key FROM %s GROUP BY grouping_key ORDER BY grouping_key", view);
+            String baseQuery = format("SELECT COUNT(*) AS a_count, grouping_key FROM %s GROUP BY grouping_key ORDER BY grouping_key", table);
+
+            MaterializedResult optimizedQueryResult = computeActual(queryOptimizationWithMaterializedView, baseQuery);
+            MaterializedResult baseQueryResult = computeActual(baseQuery);
+            assertEquals(optimizedQueryResult, baseQueryResult);
+
+            assertEquals(optimizedQueryResult.getMaterializedRows().size(), 4);
+
+            assertEquals(optimizedQueryResult.getMaterializedRows().get(0).getField(0), 4L);
```

Comment:
Try to use assertEquals() here:

https://github.com/prestodb/presto/blob/66c7001aaa240e0b816878423038854fc02d9e1c/presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueryFramework.java#L151

You can give a simple select ... from values... in the expected query.  And it will compare results from both